source/adios2/core/Engine.inl

```diff
@@ -29,7 +29,7 @@ void Engine::Write(Variable<T> &variable, const T *values)
 {
     if (m_DebugMode)
     {
-        variable.CheckDims("in call to Write");
+        variable.CheckDimsBeforeWrite("Write(" + variable.m_Name + ")");
```

Comment:
@pnorbert This might create confusion since the user might rush to look for a Write(const string &name, T* ) signature, (e.g.  Write("myvar", var.data() ); )...I would keep it as "in call to Write" since the Variable.m_Name is already printed by the exception throw by either CheckDimsCommon or CheckDimsBeforeWrite. Still, this shouldn't stop the PR.